control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go

```diff
@@ -88,6 +88,12 @@ func DetectSuboptimalMTU(ctx context.Context, mgmtClient client.Client,
 	const recommendedMinMTU = uint32(9000)
 	const ovnkOverhead = 200
 
+	if hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform &&
+		hcp.Annotations[hyperv1.ManagementPlatformAnnotation] == string(hyperv1.KubevirtPlatform) {
+		// don't create the condition in kubevirt-on-kubevirt topology
+		return nil
+	}
+
```

Comment:
This shouldn't be necessary.

Also, the mgmt platform isn't necessarily related to the MTU of the infra nodes that the VMs run on.  HCP KubeVirt as a mgmt cluster is a possible cluster topology we'd consider supporting as long as the VMs ran on metal somewhere else.

If this was done to get around the e2e condition checks, put the logic  in the e2e test to ignore mtu condition for kubevirt platform mgmt

Reply:
yes, now that we've added the annotation on the e2e execution, we should get a valid value for this condition.